aws/src/main/java/org/apache/iceberg/aws/AwsClientFactories.java

```diff
@@ -105,6 +106,7 @@ public S3Client s3() {
       return S3Client.builder()
           .httpClientBuilder(configureHttpClientBuilder(httpClientType))
           .applyMutation(builder -> configureEndpoint(builder, s3Endpoint))
+          .dualstackEnabled(s3DualStackEnabled)
```

Comment:
oh interesting, I did not know dual stack is a feature common to all AWS clients: https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/awscore/client/builder/AwsClientBuilder.html

There is a similar config in S3Configuration: https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3/S3Configuration.html

Could you take a look at if enabling one and not another would fail or cause the feature to be not enabled?

Reply:
Of course. Here is what I found:

According to the description in:
https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3/S3Configuration.Builder.html

The `dualstackEnabled()` function for S3Configuration builder is now deprecated. So I think we should use the one for all AWS clients instead.

Reply:
Ah I see. That makes sense. In that case I think this PR should be "AWS: support dual stack for AWS clients", and we introduce flag `client.dualstack-enabled` for all AWS clients.